mlkem/reduce.c

```diff
@@ -9,14 +9,40 @@
  * Description: Montgomery reduction; given a 32-bit integer a, computes
  *              16-bit integer congruent to a * R^-1 mod q, where R=2^16
  *
- * Arguments:   - int32_t a: input integer to be reduced;
- *                           has to be in {-q2^15,...,q2^15-1}
```

Comment:
Is this really no longer required? See comment in ntt.c. The code appears identical to the Kyber ref, so why can this
constraint be removed?

Reply:
I really need _weakest_ precondition on montgomery_reduce(), and the post-condition that expresses a bound on the output if that pre-condition it true, both expressed using simple static constants (i.e. not complex expressions involving real numbers...) please.

Reply:
> Is this really no longer required? See comment in ntt.c. The code appears identical to the Kyber ref, so why can this constraint be removed?

I don't see a need for the constraint in the Kyber repo, and indeed it too strict for the application in other parts of the code. Please check my bounds.

Reply:
> I really need weakest precondition on montgomery_reduce(), and the post-condition that expresses a bound on the output if that pre-condition it true, both expressed using simple static constants (i.e. not complex expressions involving real numbers...) please.

This is really a parametric spec here. We'll need to see how to best express this in CBMC, but that's not for this PR to solve.

Reply:
Noted